core/src/main/scala/org/apache/spark/deploy/history/EventLogFileCompactor.scala

```diff
@@ -221,5 +222,15 @@ private class CompactedEventLogFileWriter(
     hadoopConf: Configuration)
   extends SingleEventLogFileWriter(appId, appAttemptId, logBaseDir, sparkConf, hadoopConf) {
 
-  override val logPath: String = originalFilePath.toUri.toString + EventLogFileWriter.COMPACTED
+  override val shouldCompress = EventLogFileWriter.codecName(originalFilePath).isDefined
```

Comment:
Nice catch.

Reply:
BTW, `EventLogFileWriter.codecName(originalFilePath)` is used twice here and line 229. Could you make a private val and reuse it?

Reply:
> BTW, `EventLogFileWriter.codecName(originalFilePath)` is used twice here and line 229. Could you make a private val and reuse it?

has been fixed.